vitamins/scs_bearing_block.scad

```diff
@@ -0,0 +1,147 @@
+//
+// NopSCADlib Copyright Chris Palmer 2018
+// nop.head@gmail.com
+// hydraraptor.blogspot.com
+//
+// This file is part of NopSCADlib.
+//
+// NopSCADlib is free software: you can redistribute it and/or modify it under the terms of the
+// GNU General Public License as published by the Free Software Foundation, either version 3 of
+// the License, or (at your option) any later version.
+//
+// NopSCADlib is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+// without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with NopSCADlib.
+// If not, see <https://www.gnu.org/licenses/>.
+//
+
+//
+// SCSnUU and SCSnLUU bearing blocks
+//
+
+
+include <NopSCADlib/core.scad>
+include <NopSCADlib/vitamins/screws.scad>
+include <NopSCADlib/vitamins/linear_bearings.scad>
+
+function scs_size(type)                 = [type[4],type[6],type[5]]; //! Size of scs bracket bounding block
+function scs_hole_offset(type)          = type[2]; //! Offset of bearing hole from base of block
+function scs_block_center_height(type)  = type[6];  //! Height of the center of the block
+function scs_block_side_height(type)    = type[7];  //! Height of the side of the block, this determines the minimum screw length
+function scs_screw(type)                = type[11]; //! Screw type
+function scs_screw_separation_x(type)   = type[8];  //! Screw separation in X direction
+function scs_screw_separation_z(type)   = type[9];  //! Screw separation in Z direction
+function scs_bearing(type)              = type[14]; //! Linear bearing used
+
+
+sks_bearing_block_color = grey90;
+
+module scs_bearing_block(type) { //! Draw the specified SCS bearing block
+    vitamin(str("scs_bearing_block(", type[0], "): ", type[0], " bearing block"));
+
+    T = type[1];
+    h = type[2];
```

Comment:
These should use the accessor functions so the offset is defined in two places that have to match.